__tests__/app.test.js

```diff
@@ -16,51 +16,151 @@ beforeEach(() => {
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(topic).toHaveProperty("slug");
-                        expect(topic).toHaveProperty("description");
-                    });
-                });
+    });
+    test("Each object in the returned array should only have 2 keys", () => {
+      return request(app)
+        .get("/api/topics")
+        .expect(200)
+        .then(({ body }) => {
+          body.forEach((topic) => {
+            expect(Object.keys(topic).length).toBe(2);
+          });
         });
-        test("Each object in the returned array should only have 2 keys", () => {
-            return request(app)
-                .get("/api/topics")
-                .expect(200)
-                .then(({ body }) => {
-                    body.forEach((topic) => {
-                        expect(Object.keys(topic).length).toBe(2);
-                    });
-                });
+    });
+  });
+  describe("GET /api/articles/:article_id", () => {
+    test("Should return 200 status code", () => {
+      return request(app).get("/api/articles/1").expect(200);
+    });
+    test("Returned object should have desired keys only", () => {
+      const articleTemplate = {
+        article: {
+          article_id: 1,
+          title: "Living in the shadow of a great man",
+          topic: "mitch",
+          author: "butter_bridge",
+          body: "I find this existence challenging",
+          created_at: "2020-07-09T20:11:00.000Z",
+          votes: 100,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        },
+      };
+      return request(app)
+        .get("/api/articles/1")
+        .then(({ body }) => {
+          expect(body).toMatchObject(articleTemplate);
         });
-    })
-})
+    });
+    test("Should return status 404 and error message if article_id does not exist", () => {
+      return request(app)
+        .get("/api/articles/999")
+        .expect(404)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`We can't find that!`);
+        });
+    });
+    test("Should return status 400 and error message if article_id is invalid type", () => {
+      return request(app)
+        .get("/api/articles/bananaPancakes")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe(`That input is invalid`);
+        });
+    });
+  });
+  describe("POST /api/articles/:article_id/comments", () => {
+          const commentThis = {
+            username: "butter_bridge",
+            body: `Did you just finish all the coffee?!`,
+          }; 
+          const matchedShape = {
+            comment_id: 19,
+            body: `Did you just finish all the coffee?!`,
+            article_id: 3,
+            author: "butter_bridge",
+            votes: 0,
+            created_at: '2023-08-16T17:57:37.000Z',
```

Comment:
I'm a bit concerned that giving an explicit date value in this object might have the capacity to cause problems because in theory it will be different every time

Reply:
I was pondering this for a while - nice to know the way around this. At one point I was creating my own Date value both in the model and the tests to try avoiding a hardcoded value.... and finding my tests fail due to the millisecond differences. At least I learnt a bit more about formatting dates and the different types they come it :)